tools/common-grunt-rules.js

```diff
@@ -1,106 +1,66 @@
-// Naming this `exports` is a trick to allow this file to be compiled normally
-// and still used by commonjs-style require.
-var exports;
-(function (exports) {
-    // Compiles a module's source files, excluding tests and declarations.
-    // The files must already be available under build/.
-    function typescriptSrc(name) {
+// common-grunt-rules
+/// <reference path='../../third_party/typings/node/node.d.ts' />
+var path = require('path');
+var Rule = (function () {
+    function Rule(config) {
+        this.config = config;
+    }
+    // Grunt Jasmine target creator
+    // Assumes that the each spec file is a fully browserified js file.
+    Rule.prototype.jasmineSpec = function (name, morefiles) {
+        if (!morefiles) {
+            morefiles = [];
+        }
         return {
             src: [
-                'build/' + name + '/**/*.ts',
-                '!build/' + name + '/**/*.spec.ts',
-                '!build/' + name + '/**/*.d.ts',
-            ],
+                require.resolve('arraybuffer-slice'),
+                path.join(path.dirname(require.resolve('es6-promise/package.json')), 'dist/promise-1.0.0.js')
+            ].concat(morefiles),
             options: {
-                sourceRoot: 'build/',
-                target: 'es5',
-                comments: false,
-                noImplicitAny: true,
-                sourceMap: true,
-                declaration: false,
-                fast: 'always'
+                specs: [path.join(this.config.devBuildDir, name, '/**/*.spec.static.js')],
+                outfile: path.join(this.config.devBuildDir, name, '/SpecRunner.html'),
+                keepRunner: true
             }
         };
-    }
-    exports.typescriptSrc = typescriptSrc;
-    // Compiles a module's tests and declarations, in order to
-    // help test that declarations match their implementation.
-    // The files must already be available under build/.
-    function typescriptSpecDecl(name) {
+    };
+    // Grunt browserify target creator
+    Rule.prototype.browserify = function (filepath) {
         return {
-            src: [
-                'build/' + name + '/**/*.spec.ts',
-                'build/' + name + '/**/*.d.ts',
-            ],
+            src: [path.join(this.config.devBuildDir, filepath + '.js')],
+            dest: path.join(this.config.devBuildDir, filepath + '.static.js'),
             options: {
-                sourceRoot: 'build/',
-                target: 'es5',
-                comments: false,
-                noImplicitAny: true,
-                sourceMap: true,
-                declaration: false,
-                fast: 'always'
+                debug: true,
```

Comment:
this leaves source-map in the final file. may want to use `exorcist` to separate sourcemap to separate file.